### ISSUE TITLE
Sequence Numbers not  increases by 1

### ISSUE THREAD

**Issue by bitcodernull** [stellar/stellar-core#1541]
# Before creating an issue
 * check the [documentation](https://www.stellar.org/developers/)
 * ask for help [from the community](https://www.stellar.org/community/)

# Issue type
  want to help

# Version found
  * v9.1.0-31-g1380e88

# Your environment and setup
  * Operating System : Linux node-extend 4.4.0-62-generic #83-Ubuntu SMP Wed Jan 18 14:10:15 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
  *
# Issue description
   i  keep track of the sequence number locally
  when i broadcast the transaciton ,i  got the error: the seq error
  i find the sequence number is discontinuous
for example:  https://stellarchain.io/address/GBKIPI76UFIOK4R3HU2PGHYNMRTJ5WPCNZPHX3NJ7ABQAIKE55E2H4VG

the last transaction sequence: 69426926569127984
but the before transaction seq: 69426926569127981

**Comment by MonsieurNicolas** [comment]
Thanks for the report.

Here is what I found.

69426926569127984 occurs with this transaction
https://www.stellar.org/laboratory/#explorer?resource=transactions&endpoint=single&values=eyJ0cmFuc2FjdGlvbiI6ImNlZDhlZjJiYjljZjM4ZDE5YjMwNDBlN2QxMDFlYTQ4ZDc5MTdkZGYzY2VmMWE0ZTQxZDU4ZWRlYTM4NjJkY2UifQ%3D%3D&network=public

decoding the fee_meta_xdr yields this (in the "before" state):
lastModifiedLedgerSeq: 16223936

yet, when I look with Horizon at ledger 16223936
https://www.stellar.org/laboratory/#explorer?resource=ledgers&endpoint=single&values=eyJsZWRnZXIiOiIxNjIyMzkzNiJ9&network=public
neither of the 2 transactions in that ledger seem to have modified this account (according to the meta)

Then what I did was simply replay that ledger with stellar-core using `--catchup-at 16223937`

Here is what I got in the txhistory table:
```
txid: ba1952004d984904fb4abb4c74fd4b841fea5c152cffec1d60b6c1047e954eb9

txindex: 1

txbody: AAAAAFSHo/6hUOVyOz008x8NZGae2eJuXnvtqfgDACFE70mjAAAAZAD2p2wAAAAvAAAAAAAAAAEAAAAENTAwMAAAAAEAAAAAAAAAAQAAAABdYNu0vmaQ4tFyV4OvyIMTSID/mCQyQFSwAIFEBtHAqAAAAAFNT0JJAAAAADxxMCueZdrjARAnEMc7DlLAFoNWMF8A/mVEw1mRG2VkAAAAAAAPQkAAAAAAAAAAAUTvSaMAAABAfzxMHL3rPYvpA3I9T7va9EKf5yRiKPEnBkv7Wdha2d0m+ZrnfKQNrZeVg3Nc96nH8dpwTTAIilbN0pi/hXElBA==

txresult (TransactionResultPair):
uhlSAE2YSQT7SrtMdP1LhB/qXBUs/+wdYLbBBH6VTrkAAAAAAAAAZP////8AAAABAAAAAAAAAAH////6AAAAAA==

txmeta: AAAAAQAAAAAAAAAA
```

So... this looks like a critical bug in Horizon as failed transactions are not tracked properly.
